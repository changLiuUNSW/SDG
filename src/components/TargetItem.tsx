import React from 'react';
import styled from 'styled-components/native';

import { Target } from '../models/Target';

const Container = styled.View`
  margin-horizontal: 16;
  justify-content: center;
  align-items: center;
  border-color: #e6e9ee;
  border-radius: 8;
  border-width: 1;
  background-color: #ffff;
  margin-vertical: 8;
`;

const Text = styled.Text`
  text-align: center;
  font-size: 18;
`;

interface Props {
  target: Target;
}
const TargetItem = ({ target }: Props) => (
  <Container>
    <Text>{`${target.code}. ${target.title}`}</Text>
  </Container>
);

export default TargetItem;
