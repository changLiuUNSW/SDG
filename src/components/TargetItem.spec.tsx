import * as React from 'react';
import renderer from 'react-test-renderer';

import TargetItem from './TargetItem';

const setup = (propertiesOverride?: any) => {
  const props = Object.assign(
    {
      target: {
        code: '13.1',
        title: 'target 13'
      }
    },
    propertiesOverride
  );

  const wrapper = () => <TargetItem {...props} />;

  return {
    props,
    wrapper
  };
};

it('should render the component correctly', () => {
  const { wrapper } = setup();
  const component = renderer.create(wrapper());
  expect(component.toJSON()).toMatchSnapshot();
});
