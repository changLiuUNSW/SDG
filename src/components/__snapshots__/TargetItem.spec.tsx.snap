// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the component correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffff",
        "borderColor": "#e6e9ee",
        "borderRadius": 8,
        "borderWidth": 1,
        "justifyContent": "center",
        "marginHorizontal": 16,
        "marginVertical": 8,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 18,
          "textAlign": "center",
        },
      ]
    }
  >
    13.1. target 13
  </Text>
</View>
`;
